spot-client/src/spot-tv/ui/components/setup/select-media/mic-preview.js

```diff
@@ -101,34 +111,58 @@ export default class MicPreview extends React.PureComponent {
             return;
         }
 
-        this._destroyPreviewTrack();
+        const setLoadingPromise = new Promise(resolve =>
+            this.setState({ loading: true }, resolve));
 
-        avUtils.createLocalAudioTrack(description.deviceId)
+        setLoadingPromise
+            .then(() => this._destroyPreviewTrack())
+            .then(() => avUtils.createLocalAudioTrack(description.deviceId))
             .then(jitsiLocalTrack => {
-                this._previewTrack = jitsiLocalTrack;
+                if (jitsiLocalTrack.getDeviceId() !== description.deviceId) {
+                    jitsiLocalTrack.dispose();
 
-                this._previewTrack.on(
+                    return Promise.reject('Wrong device id received');
+                }
+
+                jitsiLocalTrack.on(
                     avUtils.getTrackEvents().TRACK_AUDIO_LEVEL_CHANGED,
                     this._updateAudioLevel
                 );
+
+                this.setState({
+                    micPreviewTrack: jitsiLocalTrack
+                });
+            })
+            .catch(error => {
+                logger.error('Mic preview failed', { error });
+            })
+            .then(() => {
+                this.setState({
+                    loading: false
```

Comment:
maybe check if the preview track is still there or this may throw errors when the component is unmounted